src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -14,14 +14,16 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
+import java.util.regex.Pattern;
 
 public class BinaryJedisCluster implements BasicCommands, BinaryJedisClusterCommands,
-    MultiKeyBinaryJedisClusterCommands, JedisClusterBinaryScriptingCommands, Closeable {
+    MultiKeyBinaryJedisClusterCommands , HighLevelBinaryJedisClusterCommands, JedisClusterBinaryScriptingCommands, Closeable {
 
   public static final short HASHSLOTS = 16384;
   protected static final int DEFAULT_TIMEOUT = 2000;
   protected static final int DEFAULT_MAX_REDIRECTIONS = 5;
-
+  protected static final Pattern SCAN_VALID_MATCH_REGEX_PATTERN = Pattern.compile("\\{[^\\{^\\}]+\\}\\*");
```

Comment:
@marcosnils  re-reading what you wrote , I see you meant changing the way we check for input matching to be based upon start/end chars. Regex will indeed be slower , I'll fix my code to behave the same as getSlot(String key) method does :smile: